### PR TITLE
UI: Fix notifications not appearing in new layout

### PR DESCRIPTION
**Pull request by JReinhold**
## What I did

This PR adds the Notifications component back into the manager's layout, that was removed by accident at some point during the development.

This is not done, here are screenshots of how it looks:

**Desktop**
image

It's unfortunate (and completely coincidental) that the default notification with the default sidebar size aligns this way, I think we should move/resize one of them.

**Mobile**

image

Should we move the notification up, or remove it completely on mobile?

The settings pages that the What's New notification takes you to is not accessible by normal means on mobile. And it's actually broken right now, the iframe containing the blogpost having a 0 px height.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This pull request has been released as version [`0.0.0-pr-24281-sha-534a6350`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-24281-sha-534a6350). Install it by pinning all your Storybook dependencies to that version.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-24281-sha-534a6350`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-24281-sha-534a6350) |
| **Triggered by** | @cdedreuille |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`fix-notifications-in-new-layout`](https://github.com/storybookjs/storybook/tree/fix-notifications-in-new-layout) |
| **Commit** | [`534a6350`](https://github.com/storybookjs/storybook/commit/534a635017fb7839d7918294235e7f232e25fa7c) |
| **Datetime** | Tue Sep 26 13:11:12 UTC 2023 (`1695733872`) |
| **Workflow run** | [6313164220](https://github.com/storybookjs/storybook/actions/runs/6313164220) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=24281`_
</details>

### CODE REVIEW

File: code/ui/manager/src/components/notifications/NotificationList.tsx [hunk 1]
```diff
@@ -4,41 +4,48 @@ import type { State } from '@storybook/manager-api';
...
+  );
+};
+
+const List = styled.div<{ placement?: CSSObject }>({
+  zIndex: 200,
+  position: 'fixed',
+  left: 20,
+  bottom: 60,
+  maxHeight: 'calc(100vh - 40px)',
+  overflowY: 'scroll',
```
Comment (JReinhold) [inline code/ui/manager/src/components/notifications/NotificationList.tsx:38]: @cdedreuille  Check out the story, especially in dark mode the scroll bar is clearly visible, do you see the same thing?

https://635781f3500dd2c49e189caf-wamswpsigz.chromatic.com/?path=/story/manager-notifications-notificationlist--single&globals=theme:dark

image

Comment (cdedreuille) [reply code/ui/manager/src/components/notifications/NotificationList.tsx:38]: Ah damn. What browser are you testing on?

Comment (cdedreuille) [reply code/ui/manager/src/components/notifications/NotificationList.tsx:38]: I'm not even sure we need that to be honest. I was just trying to make sure you can scroll even if you have a lot of notifications but if you do, this means that you can't use the sidebar :)

Comment (cdedreuille) [reply code/ui/manager/src/components/notifications/NotificationList.tsx:38]: I might just remove this overflow

Comment (cdedreuille) [reply code/ui/manager/src/components/notifications/NotificationList.tsx]: Done

**Pull request merged**